src/libxrpl/protocol/STTx.cpp

```diff
@@ -543,6 +548,32 @@ isAccountFieldOkay(STObject const& st)
     return true;
 }
 
+static bool
+invalidMPTAmountInTx(STObject const& tx)
```

Comment:
It enables a table-driven validation. TxFormats defines the amount fields that support MPT. This code validates the fields against TxFormats. Otherwise every transactor has to check for `sfFee`, which is an amount. `sfAmount` is also used in quite a few transactions.

Reply:
Every transactor (the base `Transactor`) checks that the `Fee` field is XRP and non-negative. No extra work is necessary to keep out MPT. If this code and its caller were removed, would any tests succeed that should fail? I'm guessing some different error codes would be returned, but that's fine.

Reply:
Tests will fail for the transactors, which have amount fields but don't support MPT: AMMCreate, AMMDeposit, AMMWithdraw, AMMBid, CheckCash, CheckCreate, EscrowCreate, OfferCreate, PaymentChannelCreate, PaymentChannelFund, PaymentChannelClaim, Payment, NFTokenCreateOffer, NFTokenAcceptOffer, NFTokenMint, TrustSet, XChainCommit, XChainCreateClaimID, XChainAddClaimAttestation, XChainAddAccountCreateAttestation, XChainAccountCreateCommit, XChainCreateBridge, XChainModifyBridge

Reply:
Ok, I will concede this one.